emcc.py

```diff
@@ -831,6 +831,9 @@ def get_cflags(user_args):
   if settings.WASM_WORKERS:
     cflags.append('-D__EMSCRIPTEN_WASM_WORKERS__=1')
 
+  if settings.WASMFS:
+    cflags.append('-D__EMSCRIPTEN_WASMFS__=1')
```

Comment:
I don't really like doing magic stuff like this in the driver. 

Aside from tests what kind of libraries want to compile differently?    Normally its enough to define something when that library is being built in system_libs.py.

Reply:
This is a partner request so that they can help us test out WasmFS on the bleeding edge while keeping their old stable code paths around for the time being as well. They could of course define their own analogous macro, but they preferred a built-in macro.

Reply:
I think we should just let them define their own.   

This isn't really a compiler feature, more of library that is linked in.    It think adding this could lead is to ask "why not have a macro for every setting / library we use?"

Reply:
Good question. We would want the define in one place for managing libraries between teams. If there is a #define in one place, then all teams will use that #define. Leaving it up to the teams to coordinate or to use their own defines will lead to more overhead in the transition from pthreadfs to WasmFS.

Reply:
Can you just `-DUSING_WAMFS` in the same location in your build that you already set `-sWASMFS`?

Reply:
One issue is that `-sWASMFS` today is just a linker flag, and doesn't effect compilation at all.    If we make this change that `-sWASMFS` becomes a compile time flag too, something it would be good to avoid.

Reply:
Or, it would be defined in `wasmfs.h`? But, I see your point. We don't want to include `wasmfs.h` if we're not using WasmFS.